src/examples/sparse_matrix_redecomp.cpp

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+/**
+ * @file sparse_matrix_redecomp.cpp
+ *
+ * @brief Demonstrates moving a matrix using redecomp::SparseMatrixTransfer
```

Comment:
Consider "Demonstrates moving a sparse matrix using redecomp::SparseMatrixTransfer"; it's a little pedantic, but more precise since we have multiple matrix transfers now.